### PR TITLE
Move nextgen config from product-ei repository

### PR DESCRIPTION
**Pull request by madushadhanushka**
Apply new single toml file configuration format into the micro-integrator repository.

### CODE REVIEW

File: distribution/src/scripts/micro-integrator.sh [hunk 1]
```diff
@@ -325,6 +325,7 @@ do
     -DworkerNode=false \
     -Dorg.apache.cxf.io.CachedOutputStream.Threshold=104857600 \
     -Djavax.xml.xpath.XPathFactory:http://java.sun.com/jaxp/xpath/dom=net.sf.saxon.xpath.XPathFactoryImpl \
+    -Dcarbon.new.config.dir.path="$CARBON_HOME/repository/resources/conf" \
```
Comment (madushadhanushka) [inline distribution/src/scripts/micro-integrator.sh]: move this to carbon.properties file

File: components/org.wso2.micro.integrator.server/src/main/java/org/wso2/micro/integrator/server/Main.java [hunk 2]
```diff
@@ -346,4 +353,23 @@ public URLClassLoader run() {
         }
         return initialPropertyMap;
     }
+
+    private static void handleConfiguration() {
+
+        String resourcesDir = System.getProperty(CARBON_NEW_CONFIG_DIR_PATH);
```
Comment (madushadhanushka) [inline components/org.wso2.micro.integrator.server/src/main/java/org/wso2/micro/integrator/server/Main.java]: remove this property

File: components/org.wso2.micro.integrator.server/src/main/java/org/wso2/micro/integrator/server/Main.java [hunk 3]
```diff
@@ -346,4 +353,23 @@ public URLClassLoader run() {
...
+
+        String configFilePath = System.getProperty(DEPLOYMENT_CONFIG_FILE_PATH);
+        if (StringUtils.isEmpty(configFilePath)) {
+            configFilePath = System.getProperty(LauncherConstants.CARBON_CONFIG_DIR_PATH) + File.separator +
+                             ConfigParser.UX_FILE_PATH;
+        }
+
```
Comment (madushadhanushka) [inline components/org.wso2.micro.integrator.server/src/main/java/org/wso2/micro/integrator/server/Main.java]: Add info log for config file path

Comment (madushadhanushka) [reply components/org.wso2.micro.integrator.server/src/main/java/org/wso2/micro/integrator/server/Main.java]: There is a info log in config mapper code

File: components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/Activator.java [hunk 4]
```diff
@@ -162,4 +170,24 @@ private void logServerInfo() {
...
+            configFilePath = Paths.get(System.getProperty(LauncherConstants.CARBON_CONFIG_DIR_PATH),
+                                       ConfigParser.UX_FILE_PATH).toString();
+        }
+
+        String outputDir = System.getProperty(LauncherConstants.CARBON_HOME);
+        try {
+            ConfigParser.parse(configFilePath, resourcesDir, outputDir);
+        } catch (ConfigParserException e) {
+            log.info("Error while performing configuration changes", e);
+            System.exit(1);
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/Activator.java]: Should we exit here? Or throw the exception?

File: distribution/src/conf/ciphertool.sh [hunk 5]
```diff
@@ -105,7 +105,7 @@ fi
 
 # update classpath
 CARBON_CLASSPATH=""
-for f in "$CARBON_HOME"/wso2/lib/org.wso2.ciphertool*.jar
+for f in "$CARBON_HOME"/wso2/lib/*.jar
```
Comment (sdkottegoda) [inline distribution/src/conf/ciphertool.sh]: The change should be reflected in ciphertool.bat as well, right?

File: components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/Activator.java [hunk 6]
```diff
@@ -19,13 +19,17 @@
...
 
+import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import org.osgi.framework.BundleActivator;
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.ServiceRegistration;
 //import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.server.LauncherConstants;
```
Comment (milindaperera) [inline components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/Activator.java]: We cannot use kernel constants. I think we have already defined in MI repo

File: components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/Activator.java (hunk 4 above)
Comment (madushadhanushka) [reply components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/Activator.java]: This was early discussed and decided to stop execution and show severe error

File: distribution/src/resources/config-tool/default.json [hunk 7]
```diff
@@ -0,0 +1,132 @@
+{
+  "server.enableMTOM": false,
+  "server.enableSwA": false,
+  "server.userAgent": "WSO2 Micro Integrator 1.0.0",
+  "server.serverDetails": "WSO2 Micro Integrator 1.0.0",
+  "server.synapse_config_file_path": "repository/deployment/server/synapse-configs",
+  "server.hide_admin_service_wsdl": true,
+
+  "encryption.internal_crypto_provider": "org.wso2.carbon.crypto.provider.KeyStoreBasedInternalCryptoProvider",
```
Comment (Nirothipan) [inline distribution/src/resources/config-tool/default.json]:
```suggestion
  "encryption.internal_crypto_provider": "org.wso2.micro.integrator.crypto.provider.KeyStoreBasedInternalCryptoProvider",
```

File: distribution/src/resources/config-tool/deployment-full.toml [hunk 8]
```diff
@@ -0,0 +1,535 @@
...
+# axis2.xml file
+enable_mtom = false # optional  default: false
+enable_swa = false # optional   default: false
+userAgent = "WSO2 ${product.key} ${product.version}"    #inferred
+serverDetails = "WSO2 ${product.key} ${product.version}"    #inferred
+synapse_config_file_path = "repository/deployment/server/synapse-configs"  #inferred no docs for the end user
+
+[encryption]
+#carbon.xml
+internal_crypto_provider = "org.wso2.carbon.crypto.provider.KeyStoreBasedInternalCryptoProvider" #inferred
```
Comment (Nirothipan) [inline distribution/src/resources/config-tool/deployment-full.toml]:
```suggestion
internal_crypto_provider = "org.wso2.micro.integrator.crypto.provider.KeyStoreBasedInternalCryptoProvider" #inferred
```

File: distribution/src/resources/config-tool/deployment-full.toml [hunk 9]
```diff
@@ -0,0 +1,535 @@
...
+key_password = "$ref{keystore.tls.key_password}"    # inferred
+
+##################################### Hidden properties only for Testing ###################################
+
+[[server.get_request_processor]]    # hidden property
+item = "swagger.json"
+class = "org.wso2.appcloud.api.swagger.processors.json.SwaggerJsonProcessor"
+
+[transport.local]
+sender.nonblocking.enable = false
```
Comment (Nirothipan) [inline distribution/src/resources/config-tool/deployment-full.toml]: Please add a new line here

File: distribution/src/resources/config-tool/templates/conf/axis2/axis2.xml.j2 [hunk 10]
```diff
@@ -0,0 +1,721 @@
...
+<axisconfig name="AxisJava2.0" xmlns:svns="http://org.wso2.securevault/configuration">
+
+    <!-- ================================================= -->
+    <!--                  Parameters                       -->
+    <!-- ================================================= -->
+
+    <!-- Change this to true if you want to enable hot deployment of services -->
+    <parameter name="hotdeployment" locked="false">true</parameter>
+    <!-- Change this to true if you want to enable hot update of services -->
+    <parameter name="hotupdate" locked="false">true</parameter>
```
Comment (Nirothipan) [inline distribution/src/resources/config-tool/templates/conf/axis2/axis2.xml.j2]: We need to remove above two parameters

File: distribution/src/resources/config-tool/templates/conf/synapse.properties.j2 [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+#
+##############################################
+
+{%for property, value in synapse_properties.items() %}
+{{property}} = {{value}}
+{% endfor %}
+synapse.sal.endpoints.sesssion.timeout.default=600000
+synapse.carbon.ext.tenant.info=org.wso2.micro.integrator.initializer.handler.MITenantInfoConfigurator
+mediation.flow.statistics.event.consume.interval=1000
+mediation.flow.statistics.event.clean.interval=15000
```
Comment (Nirothipan) [inline distribution/src/resources/config-tool/templates/conf/synapse.properties.j2]: Shall we add a new line at the end of the file

File: distribution/src/resources/config-tool/templates/conf/carbon.xml.j2 [hunk 12]
```diff
@@ -0,0 +1,630 @@
...
+            <Item>wsdl</Item>
+            <Class>org.wso2.micro.core.transports.util.Wsdl11Processor</Class>
+        </Processor>
+        <Processor>
+            <Item>wsdl2</Item>
+            <Class>org.wso2.micro.core.transports.util.Wsdl20Processor</Class>
+        </Processor>
+        <Processor>
+            <Item>xsd</Item>
+            <Class>org.wso2.micro.core.transports.util.XsdProcessor</Class>
```
Comment (milindaperera) [inline distribution/src/resources/config-tool/templates/conf/carbon.xml.j2]: Let's update with latest config changes here
https://github.com/wso2/micro-integrator/blob/scratch/distribution/src/conf/carbon.xml#L567

Comment (madushadhanushka) [reply distribution/src/resources/config-tool/templates/conf/carbon.xml.j2]: this has been added into the deployment toml file as custom handlers

File: distribution/src/resources/config-tool/templates/conf/axis2/axis2.xml.j2 [hunk 13]
```diff
@@ -0,0 +1,717 @@
...
+    <transportReceiver name="http" class="{{transport.http.listener.class}}">
+        {% for parameter_key,parameter_value in transport.http.listener.parameter.items() %}
+        <parameter name="{{parameter_key}}" locked="false">{{parameter_value}}</parameter>
+        {% endfor %}
+    </transportReceiver>
+    {% endif %}
+
+    <transportReceiver name="https" class="{{transport.https.listener.class}}">
+        {% for parameter_key,parameter_value in transport.https.listener.parameter.items() %}
+        <parameter name="{{parameter_key}}" locked="false">{{parameter_value}}</parameter>
```
Comment (milindaperera) [inline distribution/src/resources/config-tool/templates/conf/axis2/axis2.xml.j2]: missing <parameter name="non-blocking" locked="false">true</parameter>
for both http and https

**Pull request merged**